# Introduction to Statistics

## 1. What is Statistics?

### Understand the Concept

Statistics is the process of collecting, organizing, analyzing, interpreting, and presenting data.

It helps us understand large amounts of data and find useful information from it.

For example, if we have the marks of many students, statistics can help us find the average mark, highest mark, lowest mark, and understand how the marks are distributed.

### Demonstrate the Concept

Suppose five students scored:

60, 70, 80, 90, 100

We can calculate the average:

(60 + 70 + 80 + 90 + 100) / 5 = 80

The average gives us a simple summary of the students' marks.

In [1]:
marks = [60, 70, 80, 90, 100]

average = sum(marks) / len(marks)

print("Average marks:", average)

Average marks: 80.0


### Interpretation

The average mark is 80. This means the five students scored an average of 80 marks.

This is a simple example of how statistics can summarize a group of data into useful information.

### AI/ML Connection

Statistics is important in AI/ML because it helps us understand data before building machine learning models. It is used for analyzing data, identifying patterns, detecting outliers, understanding relationships, and evaluating model results.

## 2. Why Statistics is Important in AI/ML

### Understand the Concept

Statistics is important in AI/ML because machine learning models work with data.

Before building a model, we need to understand the data properly. Statistics helps us summarize data, identify patterns, detect outliers, understand relationships between variables, and understand how the data is distributed.

Statistics is also useful for evaluating the results of machine learning models.

### Demonstrate the Concept

Suppose we have the salaries of employees:

25000, 30000, 35000, 40000, 45000

We can use statistics to calculate the average salary and understand how much the salaries vary.

In AI/ML, similar statistical analysis can be performed on a dataset before training a machine learning model.

In [4]:
import numpy as np

salary = [25000, 30000, 35000, 40000, 45000]

mean_salary = np.mean(salary)
median_salary = np.median(salary)
standard_deviation = np.std(salary)

print("Mean salary:", mean_salary)
print("Median salary:", median_salary)
print("Standard deviation:", standard_deviation)

Mean salary: 35000.0
Median salary: 35000.0
Standard deviation: 7071.067811865475


### Interpretation

The mean and median salary are both 35000, which gives us an idea about the center of the salary data.

The standard deviation shows how much the salary values vary from the mean.

This type of statistical analysis helps us understand the dataset before using it for machine learning.

### AI/ML Use Case

Before training a machine learning model, statistics can help us understand the distribution of features, identify unusual values, check relationships between variables, and prepare the data for modeling.

## 3. Statistics vs Machine Learning

### Understand the Concept

Statistics mainly focuses on collecting, analyzing, summarizing, and interpreting data. It helps us understand what is happening in the data.

Machine Learning focuses on using data to learn patterns and make predictions or decisions.

Statistics and Machine Learning are closely related because many machine learning methods use statistical concepts.

### Demonstrate the Concept

Suppose we have house data containing area, number of bedrooms, location, and price.

Using Statistics, we can calculate the average house price and understand the relationship between area and price.

Using Machine Learning, we can use the available house data to train a model and predict the price of a new house.

### Simple Example

Statistics → "What is the average house price?"

Machine Learning → "Can we predict the price of a new house?"

In [5]:
import numpy as np

house_prices = [200000, 250000, 300000, 350000, 400000]

average_price = np.mean(house_prices)

print("Average house price:", average_price)

Average house price: 300000.0


### Interpretation

The average house price in our example is 300000.

This is an example of using statistics to summarize existing data.

In Machine Learning, we would use features such as area, number of bedrooms, and location to train a model and predict the price of a new house.

### Key Difference

| Statistics | Machine Learning |
|---|---|
| Focuses on understanding and interpreting data | Focuses on learning patterns from data |
| Used to summarize and analyze data | Used to make predictions or decisions |
| Example: Mean, variance, hypothesis testing | Example: Linear Regression, Decision Tree |

### AI/ML Use Case

Statistics helps us understand and prepare data, while Machine Learning uses the data to learn patterns and make predictions.

## 4. Descriptive Statistics

### Understand the Concept

Descriptive statistics is used to summarize and describe the data we already have.

It helps us understand the main characteristics of a dataset without making predictions about a larger population.

Common descriptive statistics include mean, median, mode, minimum, maximum, range, variance, and standard deviation.

### Demonstrate the Concept

Suppose the marks of five students are:

60, 70, 80, 90, 100

Instead of looking at every value separately, we can use descriptive statistics to find the average, minimum, maximum, and standard deviation.

This gives us a simple summary of the dataset.

In [6]:
import numpy as np

marks = [60, 70, 80, 90, 100]

print("Mean:", np.mean(marks))
print("Median:", np.median(marks))
print("Minimum:", np.min(marks))
print("Maximum:", np.max(marks))
print("Standard deviation:", np.std(marks))

Mean: 80.0
Median: 80.0
Minimum: 60
Maximum: 100
Standard deviation: 14.142135623730951


### Interpretation

The mean and median marks are 80, which tells us about the center of the data.

The minimum mark is 60 and the maximum mark is 100, which shows the lowest and highest values.

The standard deviation is approximately 14.14, which tells us how much the marks vary from the mean.

### AI/ML Use Case

Descriptive statistics is commonly used during Exploratory Data Analysis (EDA). It helps us understand the dataset, identify unusual values, understand feature distributions, and prepare the data before building a machine learning model.

## 5. Inferential Statistics

### Understand the Concept

Inferential statistics is used to make conclusions or estimates about a larger population by using data collected from a sample.

It is useful when collecting data from the entire population is difficult, expensive, or time-consuming.

### Demonstrate the Concept

Suppose a company has 10,000 customers.

Instead of asking all 10,000 customers about their satisfaction, we select 500 customers as a sample.

If 80% of the selected customers are satisfied, we can use inferential statistics to estimate the satisfaction level of the larger customer population.

### Simple Difference

Descriptive Statistics → Describes the data we collected.

Inferential Statistics → Uses sample data to make conclusions about a larger population.

In [7]:
import numpy as np

customer_satisfaction = [1, 1, 1, 0, 1, 1, 0, 1, 1, 1]

satisfaction_rate = np.mean(customer_satisfaction) * 100

print("Sample satisfaction rate:", satisfaction_rate, "%")

Sample satisfaction rate: 80.0 %


### Interpretation

The sample satisfaction rate is 80%.

This means 80% of the customers in our selected sample were satisfied.

In real-world inferential statistics, we can use a representative sample and statistical methods to estimate characteristics of the larger population. The sample result may not exactly match the population value.

### AI/ML Use Case

Inferential statistics is useful when we want to make conclusions about a larger group using limited data. It is related to areas such as hypothesis testing, confidence intervals, and statistical decision-making.

## 6. Population

### Understand the Concept

A population is the complete group of people, objects, or observations that we are interested in studying.

The population contains all the data that belongs to the study we want to perform.

### Demonstrate the Concept

Suppose we want to study the salaries of employees in a company.

If the company has 10,000 employees, the salaries of all 10,000 employees together form the population.

Population = All employees we are interested in studying.

### Example

If we want to study the marks of all students in a college, the marks of every student in that college form the population.

In [8]:
employee_salaries = [25000, 30000, 35000, 40000, 45000]

population_size = len(employee_salaries)

print("Population:", employee_salaries)
print("Population size:", population_size)

Population: [25000, 30000, 35000, 40000, 45000]
Population size: 5


### Interpretation

The list contains the salaries of all employees considered in our example.

The population size is 5 because there are 5 employees in the given population.

### AI/ML Use Case

In AI/ML, a population represents the complete group of data that we are interested in studying. Since collecting the entire population can be difficult, we often work with a sample of the population to analyze the data and build models.

## 7. Sample

### Understand the Concept

A sample is a smaller group selected from a population for analysis.

We use a sample when it is difficult, expensive, or time-consuming to collect data from the entire population.

A good sample should represent the population as closely as possible.

### Demonstrate the Concept

Suppose a company has 10,000 employees and we want to study their salaries.

Instead of collecting salary information from all 10,000 employees, we select 500 employees.

Population = 10,000 employees

Sample = 500 selected employees

The 500 employees are used to understand the salary pattern of the larger population.

In [9]:
import numpy as np

population_salaries = [
    25000, 28000, 30000, 32000, 35000,
    38000, 40000, 42000, 45000, 50000
]

sample = np.random.choice(population_salaries, size=5, replace=False)

print("Population:", population_salaries)
print("Sample:", sample)
print("Sample size:", len(sample))

Population: [25000, 28000, 30000, 32000, 35000, 38000, 40000, 42000, 45000, 50000]
Sample: [38000 30000 35000 40000 42000]
Sample size: 5


### Interpretation

The population contains all the salary values in our example.

The program randomly selects 5 values from the population and uses them as a sample.

The sample is smaller than the population, so it is easier to analyze.

### AI/ML Use Case

In AI/ML, working with the complete population may not always be practical. A representative sample can be used for data analysis, experimentation, and testing. The quality of the sample is important because a biased sample can lead to misleading conclusions.

## 8. Census

### Understand the Concept

A census means collecting data from every member of the population.

In a census, we do not select only a small sample. We collect information from the complete population.

A census can provide complete information about the population, but it may require more time, money, and resources.

### Demonstrate the Concept

Suppose a college has 1,000 students and we want to know their average marks.

If we collect the marks of all 1,000 students, it is a census.

Population = 1,000 students

Census = Data collected from all 1,000 students

In [10]:
student_marks = [65, 70, 75, 80, 85]

average_marks = sum(student_marks) / len(student_marks)

print("Marks of all students:", student_marks)
print("Average marks:", average_marks)

Marks of all students: [65, 70, 75, 80, 85]
Average marks: 75.0


### Interpretation

The program uses the marks of every student in the given population.

Since data is collected from every member of the population, this represents a census.

The average marks calculated from the complete population gives the actual population average for this example.

### AI/ML Use Case

A census can provide complete information about a dataset or population. However, in many AI/ML problems, collecting data from the entire population may not be practical, so samples are commonly used instead.

## 9. Sampling

### Understand the Concept

Sampling is the process of selecting a smaller group of observations from a larger population for analysis.

We use sampling when studying the entire population is difficult, expensive, or time-consuming.

The selected group is called a sample.

A good sampling method should try to make the sample representative of the population.

### Demonstrate the Concept

Suppose a company has 10,000 customers and wants to understand customer satisfaction.

Instead of asking all 10,000 customers, the company selects 500 customers and collects their responses.

Population = 10,000 customers

Sample = 500 selected customers

The process of selecting the 500 customers is called sampling.

In [1]:
import numpy as np

customers = np.arange(1, 101)

sample = np.random.choice(customers, size=10, replace=False)

print("Population size:", len(customers))
print("Selected sample:", sample)
print("Sample size:", len(sample))

Population size: 100
Selected sample: [ 4 35 85  5 68 74 72 18 93 43]
Sample size: 10


### Interpretation

The population contains 100 customers in our example.

The program randomly selects 10 customers from the population without selecting the same customer more than once.

The selected 10 customers represent the sample, and the process of selecting them is sampling.

### AI/ML Use Case

Sampling is useful in AI/ML when datasets are very large. A suitable sample can be used for data analysis, experimentation, and testing while reducing the time and computational resources required.

## 10. Variables

### Understand the Concept

A variable is a characteristic or value that can change between different observations.

Variables can have different values for different people, objects, or records.

For example, in a student dataset, age, marks, height, and course can be variables because their values can differ from one student to another.

### Demonstrate the Concept

Consider the following student information:

| Student | Age | Marks | Course |
|---|---:|---:|---|
| Arun | 20 | 80 | AI/ML |
| Ravi | 21 | 75 | Java |
| Priya | 20 | 90 | AI/ML |

Here, Age, Marks, and Course are variables because their values can change between students.

In [2]:
students = {
    "age": [20, 21, 20],
    "marks": [80, 75, 90],
    "course": ["AI/ML", "Java", "AI/ML"]
}

print("Age:", students["age"])
print("Marks:", students["marks"])
print("Course:", students["course"])

Age: [20, 21, 20]
Marks: [80, 75, 90]
Course: ['AI/ML', 'Java', 'AI/ML']


### Interpretation

The dataset contains three variables: age, marks, and course.

The values of these variables can differ between students.

Age and marks contain numerical values, while course contains categorical values.

### AI/ML Use Case

In AI/ML, variables represent different characteristics in a dataset. They can be used as features for a machine learning model when they provide useful information for making predictions.

## 11. Features

### Understand the Concept

A feature is an input variable that provides information to a machine learning model.

Features are used by the model to identify patterns and make predictions.

Not every variable in a dataset must be used as a feature. We usually select variables that are useful for solving the particular problem.

### Demonstrate the Concept

Suppose we want to predict the price of a house.

Our dataset contains:

- Area
- Number of bedrooms
- Location
- Age of the house
- House price

Area, number of bedrooms, location, and age can be used as features.

House price is the target value that we want to predict.

In [5]:
import pandas as pd

data = {
    "area": [1000, 1500, 2000],
    "bedrooms": [2, 3, 4],
    "age": [10, 5, 2],
    "price": [300000, 450000, 600000]
}

df = pd.DataFrame(data)

features = df[["area", "bedrooms", "age"]]
target = df["price"]

print("Features:")
print(features)

print("\nTarget:")
print(target)

Features:
   area  bedrooms  age
0  1000         2   10
1  1500         3    5
2  2000         4    2

Target:
0    300000
1    450000
2    600000
Name: price, dtype: int64


### Interpretation

The columns area, bedrooms, and age are selected as features because they provide input information for predicting the house price.

The price column is the target because it is the value we want the model to predict.

### AI/ML Use Case

Features are the input variables given to a machine learning model. The model learns patterns from these features and uses them to make predictions.

## 12. Observation

### Understand the Concept

An observation is a single record or data point in a dataset.

Each observation contains the values of different variables or features for one particular individual, object, or event.

### Demonstrate the Concept

Consider the following student dataset:

| Student | Age | Marks | Course |
|---|---:|---:|---|
| Arun | 20 | 80 | AI/ML |
| Ravi | 21 | 75 | Java |
| Priya | 20 | 90 | AI/ML |

Each row represents one observation.

For example, the first row contains the information about Arun, so it is one observation.

In [6]:
import pandas as pd

data = {
    "student": ["Arun", "Ravi", "Priya"],
    "age": [20, 21, 20],
    "marks": [80, 75, 90],
    "course": ["AI/ML", "Java", "AI/ML"]
}

df = pd.DataFrame(data)

print("Dataset:")
print(df)

print("\nFirst observation:")
print(df.iloc[0])

Dataset:
  student  age  marks course
0    Arun   20     80  AI/ML
1    Ravi   21     75   Java
2   Priya   20     90  AI/ML

First observation:
student     Arun
age           20
marks         80
course     AI/ML
Name: 0, dtype: object


### Interpretation

The dataset contains three observations because it has three rows.

Each row represents one student and contains the values of different variables for that student.

The first observation contains the information about Arun.

### AI/ML Use Case

In AI/ML datasets, each observation represents one individual data record, such as a customer, transaction, image, patient record, or house. A machine learning model learns patterns from multiple observations.

## 13. Dataset

### Understand the Concept

A dataset is a collection of related data that is organized for analysis.

A dataset usually contains multiple observations and variables.

In a tabular dataset, rows generally represent observations and columns represent variables or features.

### Demonstrate the Concept

Consider a student dataset:

| Student | Age | Marks | Course |
|---|---:|---:|---|
| Arun | 20 | 80 | AI/ML |
| Ravi | 21 | 75 | Java |
| Priya | 20 | 90 | AI/ML |

The complete table is the dataset.

Each row is an observation, and each column represents a variable.

In [7]:
import pandas as pd

data = {
    "student": ["Arun", "Ravi", "Priya"],
    "age": [20, 21, 20],
    "marks": [80, 75, 90],
    "course": ["AI/ML", "Java", "AI/ML"]
}

df = pd.DataFrame(data)

print("Dataset:")
print(df)

print("\nNumber of observations:", len(df))
print("Number of variables:", len(df.columns))

Dataset:
  student  age  marks course
0    Arun   20     80  AI/ML
1    Ravi   21     75   Java
2   Priya   20     90  AI/ML

Number of observations: 3
Number of variables: 4


### Interpretation

The DataFrame represents our dataset.

It contains 3 observations because there are 3 rows and 4 variables because there are 4 columns.

A dataset can contain different types of information depending on the problem being studied.

### AI/ML Use Case

Datasets are the foundation of machine learning. Models use datasets to learn patterns, identify relationships between features, and make predictions. The quality and relevance of the dataset can affect the performance of the model.

## 14. Quantitative Data

### Understand the Concept

Quantitative data is numerical data that represents quantities or amounts.

It can be measured or counted and can be used in mathematical calculations.

Examples include age, height, weight, salary, marks, and number of products.

### Demonstrate the Concept

Consider the following student information:

- Age = 22
- Height = 167.5 cm
- Marks = 85
- Number of subjects = 6

These values are quantitative because they represent numerical quantities.

In [8]:
import pandas as pd

data = {
    "age": [20, 21, 22],
    "height": [165.5, 170.2, 167.8],
    "marks": [80, 75, 90]
}

df = pd.DataFrame(data)

print(df)

print("\nMean age:", df["age"].mean())
print("Mean height:", df["height"].mean())
print("Mean marks:", df["marks"].mean())

   age  height  marks
0   20   165.5     80
1   21   170.2     75
2   22   167.8     90

Mean age: 21.0
Mean height: 167.83333333333334
Mean marks: 81.66666666666667


### Interpretation

Age, height, and marks are quantitative data because they contain numerical values.

Since quantitative data is numerical, we can perform mathematical and statistical calculations such as finding the mean.

### AI/ML Use Case

Quantitative data is commonly used as input features in machine learning models. For example, age, salary, height, and marks can be used as numerical features for analysis and prediction.

## 15. Qualitative Data

### Understand the Concept

Qualitative data describes categories, qualities, or characteristics rather than numerical quantities.

It is usually represented using labels or categories.

Examples include gender, city, color, department, product type, and customer feedback.

### Demonstrate the Concept

Consider the following student information:

- Course = AI/ML
- City = Coimbatore
- Department = Computer Science
- Favorite color = Blue

These values describe categories or qualities, so they are qualitative data.

In [9]:
import pandas as pd

data = {
    "student": ["Arun", "Ravi", "Priya"],
    "course": ["AI/ML", "Java", "AI/ML"],
    "city": ["Coimbatore", "Chennai", "Coimbatore"],
    "department": ["CSE", "IT", "CSE"]
}

df = pd.DataFrame(data)

print(df)

print("\nUnique courses:")
print(df["course"].unique())

print("\nUnique cities:")
print(df["city"].unique())

  student course        city department
0    Arun  AI/ML  Coimbatore        CSE
1    Ravi   Java     Chennai         IT
2   Priya  AI/ML  Coimbatore        CSE

Unique courses:
<StringArray>
['AI/ML', 'Java']
Length: 2, dtype: str

Unique cities:
<StringArray>
['Coimbatore', 'Chennai']
Length: 2, dtype: str


### Interpretation

Course, city, and department are qualitative data because they represent categories rather than numerical quantities.

The unique() function shows the different categories present in a column.

### AI/ML Use Case

Qualitative data can be useful as categorical features in machine learning. Before using categorical data in many ML models, it may need to be converted into numerical form using techniques such as label encoding or one-hot encoding.

## 16. Continuous Data

### Understand the Concept

Continuous data is numerical data that can take any value within a range.

It is usually obtained by measuring something.

Examples include height, weight, temperature, time, and distance.

Continuous data can have decimal values.

### Demonstrate the Concept

Suppose we measure the height of three students:

- 165.5 cm
- 170.2 cm
- 167.8 cm

These values are continuous data because height can take many possible values, including decimal values.

In [10]:
import pandas as pd

data = {
    "student": ["Arun", "Ravi", "Priya"],
    "height": [165.5, 170.2, 167.8],
    "weight": [55.2, 62.5, 58.7]
}

df = pd.DataFrame(data)

print(df)

print("\nAverage height:", df["height"].mean())
print("Average weight:", df["weight"].mean())

  student  height  weight
0    Arun   165.5    55.2
1    Ravi   170.2    62.5
2   Priya   167.8    58.7

Average height: 167.83333333333334
Average weight: 58.800000000000004


### Interpretation

Height and weight are continuous data because they can have decimal values and can take many possible values within a range.

The mean() function is used to calculate the average of the continuous data.

### AI/ML Use Case

Continuous data is commonly used as numerical features in machine learning.

For example, height, weight, temperature, salary, and distance can be used as features for analysis and prediction.

## 17. Discrete Data

### Understand the Concept

Discrete data is numerical data that can be counted.

It usually contains whole numbers and cannot normally have decimal values.

Examples include number of students, number of employees, number of products, and number of orders.

### Demonstrate the Concept

Suppose a shop receives:

- 10 orders
- 15 orders
- 20 orders

The number of orders is discrete data because we count the orders, and the values are whole numbers.

In [11]:
import pandas as pd

data = {
    "day": ["Monday", "Tuesday", "Wednesday"],
    "orders": [10, 15, 20],
    "customers": [8, 12, 18]
}

df = pd.DataFrame(data)

print(df)

print("\nAverage orders:", df["orders"].mean())
print("Total orders:", df["orders"].sum())

         day  orders  customers
0     Monday      10          8
1    Tuesday      15         12
2  Wednesday      20         18

Average orders: 15.0
Total orders: 45


### Interpretation

Orders and customers are discrete data because they represent counts.

The values are whole numbers because we count individual orders or customers.

### AI/ML Use Case

Discrete data can be used as numerical features in machine learning.

For example, number of purchases, number of website visits, and number of previous orders can be used to understand customer behavior and make predictions.

## 18. Structured Data

### Understand the Concept

Structured data is data that is organized in a fixed format, usually in rows and columns.

It is easy to store, search, analyze, and process because the data follows a defined structure.

Examples include Excel sheets, CSV files, SQL tables, and relational databases.

### Demonstrate the Concept

Suppose we have student information:

| Student | Age | Marks | Course |
|---|---:|---:|---|
| Arun | 20 | 80 | AI/ML |
| Ravi | 21 | 75 | Java |
| Priya | 20 | 90 | AI/ML |

This is structured data because the information is organized into rows and columns.

In [12]:
import pandas as pd

data = {
    "student": ["Arun", "Ravi", "Priya"],
    "age": [20, 21, 20],
    "marks": [80, 75, 90],
    "course": ["AI/ML", "Java", "AI/ML"]
}

df = pd.DataFrame(data)

print("Structured Data:")
print(df)

print("\nColumns:")
print(df.columns.tolist())

Structured Data:
  student  age  marks course
0    Arun   20     80  AI/ML
1    Ravi   21     75   Java
2   Priya   20     90  AI/ML

Columns:
['student', 'age', 'marks', 'course']


### Interpretation

The DataFrame is structured data because the information is arranged into rows and columns.

Each column has a specific meaning, such as age, marks, or course.

### AI/ML Use Case

Structured data is commonly used in machine learning projects.

Examples include customer records, sales data, employee information, and transaction data. Pandas and SQL are commonly used to work with structured data.

## 19. Unstructured Data

### Understand the Concept

Unstructured data is data that does not follow a fixed table or predefined structure.

It can contain different types of information such as text, images, audio, and videos.

Examples include social media posts, emails, photographs, voice recordings, and videos.

### Demonstrate the Concept

Consider these examples:

- A customer review → Text
- A product image → Image
- A voice recording → Audio
- A YouTube video → Video

These types of data do not have a simple fixed row-and-column structure, so they are considered unstructured data.

In [13]:
data = [
    "Customer review: The product is very good.",
    "Image: product_photo.jpg",
    "Audio: customer_feedback.wav",
    "Video: product_demo.mp4"
]

for item in data:
    print(item)

Customer review: The product is very good.
Image: product_photo.jpg
Audio: customer_feedback.wav
Video: product_demo.mp4


### Interpretation

The examples contain different types of information such as text, image, audio, and video.

These types of data do not follow a fixed tabular structure.

### AI/ML Use Case

Unstructured data is important in AI/ML applications.

For example, Natural Language Processing can be used for text, Computer Vision can be used for images and videos, and Speech Processing can be used for audio data.